cfme/tests/services/test_rest_services.py

```diff
@@ -41,49 +39,73 @@ def a_provider():
         pytest.skip("It's not possible to set up any providers, therefore skipping")
 
 
+def wait_for_vm_power_state(vm, resulting_state):
+    wait_for(
+        lambda: vm.power_state == resulting_state,
```

Comment:
**Optional**

Change this up a bit, providing `vm.power_state` as `func` and `resulting_state` as `fail_condition`.
From the wait_for doc:
fail_condition: An object describing the failure condition that should be tested
            against the output of func. If func() == fail_condition, wait_for continues
            to wait.